src/dex/wusdm/wusdm.ts

```diff
@@ -0,0 +1,330 @@
+        address: this.wUSDMAddress,
+        connectorTokens: [
+          {
+            decimals: 18,
+            address: this.isUSDM(tokenAddress)
+              ? this.wUSDMAddress
+              : this.USDMAddress,
+          },
+        ],
+        liquidityUSD: 1000000000, // Just returning a big number so this DEX will be preferred
+      },
+    ];
+  }
+
+  async getSimpleParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: WUSDMData,
+    side: SwapSide,
+  ): Promise<SimpleExchangeParam> {
+    const isSell = side === SwapSide.SELL;
+    const { exchange } = data;
+
+    let swapData: string;
+    if (this.isUSDM(srcToken)) {
+      swapData = this.wUSDMInterface.encodeFunctionData(
+        isSell ? WUSDMFunctions.deposit : WUSDMFunctions.mint,
+        [isSell ? srcAmount : destAmount, this.augustusAddress],
+      );
+    } else {
+      swapData = this.wUSDMInterface.encodeFunctionData(
+        isSell ? WUSDMFunctions.redeem : WUSDMFunctions.withdraw,
+        [
+          isSell ? srcAmount : destAmount,
+          this.augustusAddress,
+          this.augustusAddress,
+        ],
+      );
+    }
+
+    return this.buildSimpleParamWithoutWETHConversion(
+      srcToken,
+      srcAmount,
+      destToken,
+      destAmount,
+      swapData,
+      exchange,
+      undefined,
+      undefined,
+      undefined,
+      isSell && this.isUSDM(destToken),
+    );
+  }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: WUSDMData,
+    side: SwapSide,
+    _: Context,
+    executorAddress: Address,
+  ): DexExchangeParam {
+    const isSell = side === SwapSide.SELL;
+    const { exchange } = data;
+
+    let swapData: string;
+    if (this.isUSDM(srcToken)) {
+      swapData = this.wUSDMInterface.encodeFunctionData(
+        isSell ? WUSDMFunctions.deposit : WUSDMFunctions.mint,
+        [isSell ? srcAmount : destAmount, recipient],
+      );
+    } else {
+      swapData = this.wUSDMInterface.encodeFunctionData(
+        isSell ? WUSDMFunctions.redeem : WUSDMFunctions.withdraw,
+        [isSell ? srcAmount : destAmount, recipient, executorAddress],
+      );
+    }
+
+    return {
+      needWrapNative: this.needWrapNative,
+      dexFuncHasRecipient: true,
+      exchangeData: swapData,
+      targetExchange: exchange,
+      returnAmountPos: isSell
+        ? extractReturnAmountPosition(
+            this.wUSDMInterface,
+            this.isUSDM(srcToken)
+              ? WUSDMFunctions.deposit
+              : WUSDMFunctions.redeem,
+          )
+        : undefined,
+      skipApproval: isSell && this.isUSDM(destToken),
+    };
+  }
+
+  // Encode params required by the exchange adapter
+  // V5: Used for multiSwap, buy & megaSwap
+  // V6: Not used, can be left blank
+  // Hint: abiCoder.encodeParameter() could be useful
+  getAdapterParam(
```

Comment:
@JackieJoo I think we can left this method empty since we are not going to support this integration on v5 I assume.